product_variant_sale_price/tests/test_product_product.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo.tests.common import TransactionCase
+
+
+class TestProductVariantPrice(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.template = cls.env["product.template"]
+        cls.product_product = cls.env["product.product"]
+        cls.attribute = cls.env["product.attribute"]
+        cls.attribute_value = cls.env["product.attribute.value"]
+
+        cls.att_color = cls.attribute.create({"name": "color_test"})
+
+        cls.att_color_blue = cls.attribute_value.create(
+            {"name": "Blue", "attribute_id": cls.att_color.id}
+        )
+        cls.att_color_red = cls.attribute_value.create(
+            {"name": "Red", "attribute_id": cls.att_color.id}
+        )
+
+        cls.uom_unit = cls.env.ref("uom.product_uom_unit")
+        cls.uom_dozen = cls.env.ref("uom.product_uom_dozen")
+
+        cls.product_template = cls.template.create(
+            {
+                "name": "Product Template",
+                "list_price": 1500.00,
+                "attribute_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "attribute_id": cls.att_color.id,
+                            "value_ids": [
+                                (6, 0, (cls.att_color_blue + cls.att_color_red).ids)
+                            ],
+                        },
+                    )
+                ],
+                "uom_id": cls.uom_unit.id,
+            }
+        )
+
+        cls.product_blue = cls.product_template.product_variant_ids.filtered(
+            lambda x: x.product_template_attribute_value_ids.product_attribute_value_id
+            == cls.att_color_blue
+        )
+        cls.product_red = cls.product_template.product_variant_ids.filtered(
+            lambda x: x.product_template_attribute_value_ids.product_attribute_value_id
+            == cls.att_color_red
+        )
+
+    def test_post_init_hook(self):
+        from ..hooks import set_sale_price_on_variant
+
+        self.product_template.product_variant_ids.write({"fix_price": 0.0})
+        # Take account price extra
+        self.product_blue.product_template_attribute_value_ids.write(
+            {"price_extra": 100.00}
+        )
+        self.product_red.product_template_attribute_value_ids.write(
+            {"price_extra": 200.00}
+        )
+
+        # Flush the records to DB before direct SQL
+        self.product_blue.flush()
+        self.product_red.flush()
```

Comment:
Check https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

flush is deprecated